app/cli.php

```diff
@@ -3,30 +3,78 @@
 require_once __DIR__ . '/init.php';
 require_once __DIR__ . '/controllers/general.php';
 
-use Utopia\App;
+use Appwrite\CLI\Tasks;
 use Utopia\CLI\CLI;
-use Utopia\CLI\Console;
 use Utopia\Database\Validator\Authorization;
+use Utopia\Platform\Service;
+use Utopia\App;
+use Utopia\CLI\Console;
+use Utopia\Cache\Adapter\Redis as RedisCache;
+use Utopia\Cache\Cache;
+use Utopia\Database\Adapter\MariaDB;
+use Utopia\Database\Database;
 
 Authorization::disable();
 
-$cli = new CLI();
-
-include 'tasks/doctor.php';
-include 'tasks/maintenance.php';
-include 'tasks/install.php';
-include 'tasks/migrate.php';
-include 'tasks/sdks.php';
-include 'tasks/specs.php';
-include 'tasks/ssl.php';
-include 'tasks/vars.php';
-include 'tasks/usage.php';
-
-$cli
-    ->task('version')
-    ->desc('Get the server version')
-    ->action(function () {
-        Console::log(App::getEnv('_APP_VERSION', 'UNKNOWN'));
-    });
+CLI::setResource('register', fn()=>$register);
+
+$attempts = 0;
+$max = 10;
+$sleep = 1;
+
+do {
+    try {
+        $attempts++;
+        $db = $register->get('db');
+        $redis = $register->get('cache');
+        break; // leave the do-while if successful
+    } catch (\Exception $e) {
+        Console::warning("Database not ready. Retrying connection ({$attempts})...");
+        if ($attempts >= $max) {
+            throw new \Exception('Failed to connect to database: ' . $e->getMessage());
+        }
+        sleep($sleep);
+    }
+} while ($attempts < $max);
+
+CLI::setResource('db', fn () => $db);
+CLI::setResource('cache', fn () => $redis);
+
+CLI::setResource('dbForConsole', function ($db, $cache) {
+    $cache = new Cache(new RedisCache($cache));
+
+    $database = new Database(new MariaDB($db), $cache);
+    $database->setDefaultDatabase(App::getEnv('_APP_DB_SCHEMA', 'appwrite'));
+    $database->setNamespace('_console');
+
+    return $database;
+}, ['db', 'cache']);
+
+/** @var InfluxDB\Client $client */
+$client = $register->get('influxdb');
```

Comment:
We can have this logic inside the `setResource` callback instead of the global scope.

Reply:
This was avoided in `init.php` i was hoping there's a reason so I kept it in global scope as well.

Reply:
updated to put it in the callback